controllers/alphacontrollers/alert_controller.go

```diff
@@ -954,35 +964,50 @@ func flattenDaysOfWeek(daysOfWeek []alerts.DayOfWeek, daysOffset int32) []coralo
 	return result
 }
 
-func flattenNotificationGroups(notificationGroups []*alerts.AlertNotificationGroups) []coralogixv1alpha1.NotificationGroup {
+func flattenNotificationGroups(ctx context.Context, notificationGroups []*alerts.AlertNotificationGroups) ([]coralogixv1alpha1.NotificationGroup, error) {
 	result := make([]coralogixv1alpha1.NotificationGroup, 0, len(notificationGroups))
+	errors := make([]error, 0)
 	for _, ng := range notificationGroups {
-		notificationGroup := flattenNotificationGroup(ng)
+		notificationGroup, errs := flattenNotificationGroup(ctx, ng)
+		errors = append(errors, errs...)
 		result = append(result, notificationGroup)
 	}
-	return result
+
+	if len(errors) > 0 {
+		var errStr string
+		for _, err := range errors {
+			errStr += err.Error() + "\n"
+		}
+		return result, fmt.Errorf(errStr)
+	}
+
+	return result, nil
```

Comment:
nit:

Just in case you think this would be better, there are a few libraries that deal with concatenating multiple errors, for example:
* https://pkg.go.dev/github.com/hashicorp/go-multierror
* https://pkg.go.dev/errors

We could get rid of this if/for in case it sounds useful to you